cxx/emissions/gaussian.hh

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include "emissions/base.hh"
+
+class GaussianEmission : public Emission<double> {
+ public:
+  // We use an Inverse gamma conjugate prior, so our hyperparameters are
+  double alpha = 1.0;
+  double beta = 1.0;
+
+  // Sufficient statistics of observed data.
+  double var = 0.0;
+
+  GaussianEmission() {}
+
+  void incorporate(const std::pair<double, double> &x) {
+    ++N;
+    double y = x.second - x.first;
+    var += y * y;
+  }
+
+  void unincorporate(const std::pair<double, double> &x) {
+    --N;
+    double y = x.second - x.first;
+    var -= y * y;
+  }
+
+  double logp(const std::pair<double, double> &x) const {
+    double y = x.second - x.first;
+    // TODO(thomaswc): This.
+    return 0.0;
+  }
+
+  double logp_score() const {
+    // TODO(thomaswc): This.
+    return 0.0;
+  }
+
+  double sample_corrupted(const double& clean, std::mt19937 *prng) {
+    double smoothed_var = var;  // TODO(thomaswc): Fix
+    std::normal_distribution<double> d(clean, sqrt(smoothed_var));
+    return d(*prng);
+  }
```

Comment:
I think this is a good / totally sensible emission model for reals--assuming that independent and zero-mean Gaussian noise with unknown variance has been added. In the future we may want to allow for non-zero-mean noise (so we would also store a mean of the x.second-x.first values here). This way we could e.g. cluster shopping websites by whether they systematically add more or less to some base price for an object.

Reply:
Cool.  Another emission model I've been thinking about was zero-mean Gaussian noise with variance proportional to the clean value.